### ISSUE TITLE
Feature suggestion: More than one geometry column

### ISSUE THREAD

**Issue by PabloIsCoding** [geopandas/geopandas#1171]
Hi, I'm a new user of geopandas and it might be a bit too bold to suggest changes already but I think this package has a huge potential and want to contribute with an idea. I've noticed that when you `sjoin` two GDF's there will be just one geometry column, the one of the left GDF. You also need to put into a separate DF anything you want to plot because of this restriction of having just one geometry column.

But wouldn't it be better that GDF's were regular DF's with a special attribute called geometry that contained a DF with all the different geometries? 

Example. Say that I have 'counties', 'streets' and 'stores'. Then this geometry DF would have three columns called 'counties', 'streets' and 'stores'. Then if I just want to plot counties and stores I would do: `df.geometry[['counties', 'stores']].plot()` and be done with it. 

Does this make sense? 

Hope it helps and thanks for the good work you did here!

**Comment by martinfleis** [comment]
Hi,
couple of thoughts on it.

1) you can have multiple columns containing geometry, active as GDF geometry can be just one at a time. You can switch between them using `gdf.set_geometry()` (http://geopandas.org/data_structures.html?highlight=set_geometry#geodataframe)

2) One row of GeoDataFrame represents one feature. In your example, counties, streets and stores are different features, so for each if them you have only one geometry per row, you do not need multiple columns to do that. If you want you can have all of them as a part of one GeoDataFrame already. But it feels like a misconception of GeoDataFrame logic.

3) the same applies to your `sjoin` point. Due to the nature of sjoin, it does not make much sense to me to return geometries as is it spatial join of attributes of one geometry to another geometry. I can't think of user case when I would want to return geometry as a result of sjoin, can you?

4) your plotting example would become really complex very quickly as you need to specify plotting args for `counties` as well as `stores` (and probably more), so it, in fact, complicates the situation rather than simplifies. 

To sum up, I don't think these are changes I would support. But thanks for the ideas anyway!

**Comment by PabloIsCoding** [comment]
Okay I think set_geometry solves this already! Thanks

On your other points: the example was about having three gdf's (one for counties, one for streets and one for stores) each with it's own geometry (presumably polygons for the first two and points for the stores) and preserving all that information after the sjoin separately. I guess I think about points, polygons and such as just one more feature, not the central information describing an object (?). 

But to be honest storing everything with sjoin scales very poorly so it makes sense to have them in separate gdf's if possible, so I guess this is a non-issue. I'm closing this!
